### PR TITLE
Allow overriding the iOS platform destination for RN E2E testing

### PR DESCRIPTION
**Pull request by mokagio**
## What?

The core change in this PR is adding an environment variable that can be set to change the `xcodebuild` `-destination` parameter of one of the tests commands. This will unblock creating the bundle for SauceLabs on Buildkite, see https://github.com/wordpress-mobile/gutenberg-mobile/pull/6134

The other commits are tangential changes related to making the SauceLabs builds in CI possible or faster. Refer to the expanded commit messages for more details.

## Why?
The context of these changes is that we are incrementally moving from CircleCI to Buildkite as the CI provider for `gutenberg-mobile`. Unfortunately, there are some implicit dependencies on the CI provider and between that and this repository in the automation scripts, which is why these changes are necessary.

## How?
Moved the `test:e2e:build-app:ios` command from a long inline string in the `package.json` to a dedicated script and made it check the environment for `RN_EDITOR_E2E_IOS_DESTINATION` to set the `-destination` parameter.

## Testing Instructions

See https://github.com/wordpress-mobile/gutenberg-mobile/pull/6138 which runs the gutenberg-mobile CI against this PR.

### Testing Instructions for Keyboard

N.A.

## Screenshots or screencast 
N.A.

### CODE REVIEW

File: packages/react-native-editor/ios/.ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.7.5
+2.7.4
```
Comment (mokagio) [inline packages/react-native-editor/ios/.ruby-version:1]: That is the version supported by our CI. We could of course install a new one at runtime, but that adds 4 to 5 minutes to the build.

image


Worth noting that version 2.7.4 and 2.7.5 both are past support. We ought to upgrade to version 3.x, but that's something we might want to do in a dedicated PR.

File: packages/react-native-editor/ios/GutenbergDemo.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -702,7 +702,6 @@
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CLANG_ENABLE_MODULES = YES;
-				CODE_SIGN_IDENTITY = "iPhone Developer";
```
Comment (mokagio) [inline packages/react-native-editor/ios/GutenbergDemo.xcodeproj/project.pbxproj:705]: This is redundant. Removing it makes the project default to "Apple Developer" which is good enough for what we need here.

File: packages/react-native-editor/bin/build_e2e_ios_app [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	-destination '$DESTINATION'"
+
+# Check if xcbeautify is available
+if command -v xcbeautify >/dev/null 2>&1; then
+	# Pipe the output to xcbeautify
+	eval "$XCODEBUILD_CMD" | xcbeautify
+else
+	# Run xcodebuild without piping the output
+	eval "$XCODEBUILD_CMD"
+fi
```
Comment (mokagio) [inline packages/react-native-editor/bin/build_e2e_ios_app:27]: I extracted the `xcodebuild` call for two reasons:

1. Allow overriding the `-destination` parameter
2. Use `xcbeautify` to parse the output to make local debugging easier

**Review by geriux — APPROVED** [review]
LGTM! I've tested running the E2E tests locally and it worked as expected. 🚀

**Pull request merged**